docs/proposals/volume-hostpath-qualifiers.md

```diff
@@ -0,0 +1,96 @@
+| `new-file` | Like file, but if nothing exist an empty file is created instead | In general, bindmounting the parent directory of the file you intend to create addresses this usecase |
+
+### Volume SubPaths
+
+Volumes may also have a `subPath` specified, which similarly may or may not exist. I similarly propose that the [v1.VolumeMount](https://github.com/kubernetes/kubernetes/blob/d26b4ca2859aa667ad520fb9518e0db67b74216a/pkg/api/types.go#L886-L897) type be augmented with a `SubPathType` field which the exact same semantics as above.
+While I propose exposing the same set of possible values for consistency, in reality a `SubPath` is useful primarily under a network mount. Because of this, the `device` and `socket` values would likely not be useful.
+
+I do not have any examples of concrete examples from the Kubernetes project for this, in part because there are very few examples of `SubPath` to begin with.
+However, each of the use-cases described for Host Volumes also could conceivably occur as subdirectories of a network mount using this feature, and thus still offer insight into why it would be of value.
+
+<!-- uncomment if this comes up :) An argument against providing this feature for SubPaths is that it might not increase correctness as much; while adding such qualifications to Host volumes allows a pod to express a dependency that the node may not meet, adding it to subPath allows the pod to express an assumption about a given filesystem it's mounting. This assumption is not node specific and, when it fails, likely means the given volume was not pre-populated/provisioned appropriately. It's arguable whether this sort of failure is as likely or as important to handle well. -->
+
+
+### Why not as part of any other volume types?
+
+This feature does not make sense for any of the other volume types simply because all of the other types are already fully qualified. For example, NFS volumes are known to always be in existence else they will not mount. Similarly, EmptyDir volumes will always exist as a directory.
+
+Only the HostVolume and SubPath means of referencing a path have the potential to reference arbitrary incorrect or nonexistent things without erroring out.
+
+
+## Kubelet implementation
+
+It's proposed that prior to starting a pod, the Kubelet validates that the given path meets the qualifications of its type. Namely, if the type is `auto` the Kubelet will create an empty directory if none exists there, and for each of the others the Kubelet will perform the given validation prior to running the pod. This validation might be done by a volume plugin, but further technical consideration (out of scope of this proposal) is needed.
+
+
+## Possible concerns
+
+### Containerized Kubelet
+
+A containerized kubelet would have difficulty creating directories. The implementation will likely respect the `containerized` flag, or similar, allowing it to either break out or be "/rootfs/" aware and thus operate as desired.
```

Comment:
Containerized kubelets effectively are required to have access to host rootfs in practice (that's the only variant we support in production at Red Hat), so seems ok.